tests/test_quickbooks_interrupted_sync.py

```diff
@@ -0,0 +1,168 @@
+        #   accounts: synced records successfully
+        #   payments: remaining to sync
+        state = {
+            "currently_syncing": "bill_payments",
+            "bookmarks": { "accounts": {"LastUpdatedTime": "2021-08-10T01:10:04-07:00"}}
+        }
+
+        # Set state for 2nd sync
+        menagerie.set_state(conn_id, state)
+
+        # Run sync after interruption
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+        synced_records_interrupted_sync = runner.get_records_from_target_output()
+        record_count_by_stream_interrupted_sync = runner.examine_target_output_file(
+            self, conn_id, self.expected_streams(), self.expected_primary_keys())
+
+        # Verify tap and target exit codes
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        final_state = menagerie.get_state(conn_id)
+        currently_syncing = final_state.get('currently_syncing')
+
+        # Checking resuming the sync resulted in a successfully saved state
+        with self.subTest():
+
+            # Verify sync is not interrupted by checking currently_syncing in the state for sync
+            self.assertIsNone(currently_syncing)
+
+            # Verify bookmarks are saved
+            self.assertIsNotNone(final_state.get('bookmarks'))
+
+            # Verify final_state is equal to uninterrupted sync's state
+            # (This is what the value would have been without an interruption and proves resuming succeeds)
+            self.assertDictEqual(final_state, full_sync_state)
+
+        # Stream level assertions
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                
+                # Gather actual results
+                full_records = [message['data'] for message in synced_records_full_sync.get(stream, {}).get('messages', [])]
+                interrupted_records = [message['data'] for message in synced_records_interrupted_sync.get(stream, {}).get('messages', [])]
+                interrupted_record_count = record_count_by_stream_interrupted_sync.get(stream, 0)
+
+                # Final bookmark after interrupted sync
+                final_stream_bookmark = final_state['bookmarks'][stream]['LastUpdatedTime']
+
+                # NOTE: All streams are INCREMENTAL streams for this Tap.
+
+                # Verify final bookmark saved the match formatting standards for resuming sync
+                self.assertIsNotNone(final_stream_bookmark)
+                self.assertIsInstance(final_stream_bookmark, str)
+                self.assertIsDateFormat(final_stream_bookmark, "%Y-%m-%dT%H:%M:%S%z") # Bookmark is being saved with timezone
+
+                if stream == state['currently_syncing']:
+
+                    # Assign the start date to the interrupted stream
+                    interrupted_stream_datetime = start_date_datetime
```

Comment:
This should be using an actual bookmark for the interrupted stream instead of using the start-date so that it proves that we're resuming from the point where we left off, rather than that a stream can be synced from it's start-date (which is probably already covered under other tests).

Reply:
Agreed. Made the changes.

Reply:
lgtm 👍